packages/runtime-core/src/scheduler.ts

```diff
@@ -88,6 +88,7 @@ export function invalidateJob(job: SchedulerJob) {
   const i = queue.indexOf(job)
   if (i > -1) {
     queue.splice(i, 1)
+    if (flushIndex > 0) flushIndex -= 1
```

Comment:
Reset `flushIndex` to -1 will lead to some components double update, see code.

https://github.com/vuejs/vue-next/blob/b81d279e40c3d32a3740ea20d7e182690294f117/packages/runtime-core/src/scheduler.ts#L211-L219

Reply:
I update my solution.